web/src/hotkey.js

```diff
@@ -588,14 +588,6 @@ export function process_enter_key(e) {
     }
 
     if (processing_text()) {
-        if (stream_list.searching()) {
-            // This is sort of funny behavior, but I think
-            // the intention is that we want it super easy
-            // to close stream search.
-            stream_list.clear_and_hide_search();
-            return true;
-        }
```

Comment:
Hmm, so this is about `Esc` when you have something typed in stream search. Are you sure we don't want that to still clear it, even though it can't be hidden now? I feel like we might have other inputs that would be useful guidance.